lib/reverse_words.rb

```diff
@@ -1,6 +1,30 @@
 # A method to reverse each word in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n), where n is the length of the string 
+# Space complexity: O(1)
+
 def reverse_words(my_words)
-  raise NotImplementedError
+  return false if my_words == nil || my_words.length == 0  
+  word_start = 0
+  word_end = 0
+  i = 0
+  
+  while i < my_words.length
+    while  i < my_words.length && my_words[i] == " "
+      i += 1
+    end
+    word_start = i
+
+    while i < my_words.length && my_words[i] != " "
+      i += 1
+    end 
+    word_end = i - 1
+    while word_start < word_end
```

Comment:
Style nit: could we refactor this into a helper method? (Perhaps we can even use a method we wrote for an earlier assignment? 🙂)